ERCS/erc-7683.md

```diff
@@ -72,6 +72,100 @@ Cross-chain execution systems implementing this standard SHOULD use a sub-type t
 
 All sub-types SHOULD be registered at github.com/erc-7683/order-subtypes to encourage sharing of sub-types based on their functionality.
 
+### Decoding cross-chain orders
+
+A compliant cross-chain order type MUST be convertible into the `ResolvedCrossChainOrder` struct. This means that the `orderData` must be decoded into parameters that are used to populate the `ResolvedCrossChainOrder` struct. The decoded `orderData` SHOULD also contain data to construct a sub-type, which can be used for further functionality such as cross-chain calldata execution. It is the responsibility of the `user` and the `filler` to ensure that the `settlementContract` supports their order's contained sub-type. For example, there might be a sub-type called `Message`, which is defined by the following structs:
+
+```solidity
+// The Message subtype allows ERC7683 intents to carry calldata that is executed on a target contract on the destination chain. The settlement contract that the filler interacts with on the destination chain will decode the message into smart contract calls and execute the calls within the filler's `fill()` transaction.
+
+// The Message contains calls that the user wants executed on the destination chain.
+// The target is a contract on the destination chain that the settlement contract will attempt to send callData and value to.
+struct Calls {
+  address target;
+  bytes callData;
+  uint256 value;
+}
+
+// A fallbackRecipient can be set by the user in case sending the callData to the target reverts and the user still wants to guarantee that they receive their value to an EOA
+// on the destination chain.
+struct Message {
+  Calls[] calls;
+  address fallbackRecipient;
+}
+```
+
+The `Message` sub-type is designed to be used by a 7683 user to incentivize a filler to to execute arbitrary calldata on a target destination chain contracton the user's behalf. For example, the settlement contract might decode the `orderData` containing the message information as follows:
+
+```solidity
+function fill(bytes32 orderId, bytes calldata originData, bytes calldata fillerData) public {
+	// Input[] userInputs;
+	// Output[] maxOutputs;
+	// Output[] minFillerOutputs;
+	(
+		address user,
+		uint32 originChainId,
+		uint32 initiateDeadline,
+		uint32 fillDeadline,
+		Input[] memory userInputs,
+		Output[] memory maxOutputs,
+ 		Output[] memory minFillerOutputs,
```

Comment:
I think this downplays one of the really nice advantages of the structure of 7683: we typically don't need inputs or filler outputs on the destination chain. We also probably don't need an _array_ of outputs, since we know what chain we're doing the fill on (so only one of the elements is really being executed here).

Reply:
Hm fair point that's the nice thing about `originData` actually. Does this make sense? https://github.com/across-protocol/ERCs/pull/2/commits/123b90a1cecfec9f15984b4630abf8b98707e676